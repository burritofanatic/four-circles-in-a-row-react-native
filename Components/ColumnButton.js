import React, {Component} from 'react';
import {View, StyleSheet, TouchableHighlight, Dimensions} from 'react-native';

import * as Constants from '../Constants';

const width = Dimensions.get('window').width;
const height = Dimensions.get('window').height;

const WIDTH_OF_BOARD_TO_SCREEN = Constants.WIDTH_OF_BOARD_TO_SCREEN;
const NUMBER_OF_ROWS = Constants.NUMBER_OF_ROWS;

/*
 * The ColumnButton acts as an overlay that detects which column the user wants
 * to put his or her button in. The offset is passed in as a prop for the layout
 * along with the column value so that we can pass the result back.
 */
class ColumnButton extends Component {
  constructor(props) {
    super(props);
  }

  handleClick = () => {
    this.props.onColumnClicked(this.props.column);
  }

  onPressIn = () => {
    this.props.onColumnPressedIn(this.props.column);
  }

  render() {
    return (
      <TouchableHighlight style={[this.props.offset, styles.genericLayout]}
                          disabled={this.props.isGameOver}
                          onPress={this.handleClick}
                          onPressIn={this.onPressIn}>
        <View style={[this.props.offset, styles.genericLayout]} />
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  genericLayout: {
    position: 'absolute',
    backgroundColor: 'transparent',
    flexDirection: 'row',
    width: (width * WIDTH_OF_BOARD_TO_SCREEN) / NUMBER_OF_ROWS,
    top: 0,
    bottom: 0,
    zIndex: 10,
    opacity: 0.5
  }
});

export default ColumnButton;
