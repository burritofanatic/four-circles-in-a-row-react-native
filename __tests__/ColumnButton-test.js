import React from 'react';
import ColumnButton from '../Components/ColumnButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
        .create(<ColumnButton column={0} isGameOver={true}/>)
        .toJSON();
  expect(tree).toMatchSnapshot();
})
