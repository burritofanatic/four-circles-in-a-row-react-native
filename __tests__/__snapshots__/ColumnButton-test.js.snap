// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flexDirection": "row",
        "opacity": 0.5,
        "position": "absolute",
        "top": 0,
        "width": 85.71428571428571,
        "zIndex": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flexDirection": "row",
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 85.71428571428571,
          "zIndex": 10,
        },
      ]
    }
  />
</View>
`;
